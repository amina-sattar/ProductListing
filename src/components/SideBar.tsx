import  React, {useState} from 'react';
import Drawer from '@mui/material/Drawer';
import Toolbar from '@mui/material/Toolbar';
import List from '@mui/material/List';
import Divider from '@mui/material/Divider';
import ListItem from '@mui/material/ListItem';
import ListItemText from '@mui/material/ListItemText';
import ListSubheader from '@mui/material/ListSubheader';
import ListItemButton from '@mui/material/ListItemButton';
import { ChildCategory } from '../types';

const drawerWidth = 240;

type Props = {
  categories: ChildCategory[];
};
export default function PermanentDrawerLeft(props: Props) {
  const { categories } = props;
  const [selectedCategory, setSelectedCategory] = useState('');
  return (
    <Drawer
      sx={{
        width: drawerWidth,
        flexShrink: 0,
        '& .MuiDrawer-paper': {
          width: drawerWidth,
          boxSizing: 'border-box',
        },
      }}
      variant="permanent"
      anchor="left"
    >
      <Toolbar />
      <Divider />
      <List
        component="nav"
        aria-labelledby="nested-list-subheader"
        sx={{ paddingTop: '20px' }}
        subheader={
          <ListSubheader component="div" id="nested-list-subheader">
            Kategorien
          </ListSubheader>
        }
      >
        {categories.map(({ name }) => (
            <ListItemButton 
              key={name} 
              selected={selectedCategory === name} 
              onClick={()=> setSelectedCategory(name)} 
              sx={{':hover': {color: "home24.light"}}}>
              <ListItem>
                <ListItemText primary={name}/>
              </ListItem>
            </ListItemButton>
          ))}
      </List>
    </Drawer>
  );
}