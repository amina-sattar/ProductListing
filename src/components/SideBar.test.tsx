import React from 'react';
import renderer from 'react-test-renderer';
import { ChildCategory } from '../types';
import SideBar  from './SideBar';
import {render } from '@testing-library/react';

const dummyCategories: ChildCategory[] = [{
  name: 'Test Category',
  urlPath:'/test-category'
}]

describe('test side bar', () => {
  test('snapshot test', () => {
    const component = renderer.create(
      <SideBar categories={dummyCategories}/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  test('test category name exist', () => {
    const { getByText } = render(<SideBar  categories={dummyCategories}/>);
    const searchElement = getByText(/test category/i);
    expect(searchElement).toBeInTheDocument();
  })
})
