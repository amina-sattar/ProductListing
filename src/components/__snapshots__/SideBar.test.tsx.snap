// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test side bar snapshot test 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked css-1f2xuhi-MuiDrawer-docked"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    />
    <hr
      className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <nav
      aria-labelledby="nested-list-subheader"
      className="MuiList-root MuiList-padding MuiList-subheader css-1qpl6dz-MuiList-root"
    >
      <div
        className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-uob957-MuiListSubheader-root"
        id="nested-list-subheader"
      >
        Kategorien
      </div>
      <div
        className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1ghv7cd-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          disabled={false}
        >
          <div
            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Test Category
            </span>
          </div>
        </li>
      </div>
    </nav>
  </div>
</div>
`;
